lib/src/util.dart

```diff
@@ -1,7 +1,13 @@
 import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:flutter/foundation.dart';
 
-dynamic getSubpath(Map<String, dynamic> root, String path) {
-  return _getSubpath(root, path.split('/'));
+dynamic getSubpath(Map<String, dynamic> root, String path,
+    {bool isCollectionGroup = false}) {
+  final pathSegments = path.split('/');
+  if (isCollectionGroup) {
```

Comment:
This function  seems to be doing something very different from getSubpath: getSubpath traverses a tree, and returns the right node. buildSubpathForCollectionGroup replicates a tree, and keeps only branches that contain `collectionId`. Instead of adding a parameter to getSubpath and calling one function or the other, should we make them separate functions instead?